internal/chain/state.go

```diff
@@ -401,6 +401,10 @@ func unmarshalRecord(obj *state.Object) (state.Chain, error) {
 		record = new(protocol.KeyPage)
 	case types.ChainTypeKeyBook:
 		record = new(protocol.KeyBook)
+	case types.ChainTypeDataAccount:
+		record = new(protocol.DataAccount)
+	case types.ChainTypeLiteDataAccount:
+		record = new(protocol.LiteDataAccount)
 	default:
 		return nil, fmt.Errorf("unrecognized chain type %v", header.Type)
 	}
```

Comment:
IIRC, this switch statement exists in at least three places: here, API v1, and API v2.

Maybe we should add `func NewChain(typ types.ChainType) (state.Chain, error)` to `package protocol`<sup>1</sup>? Then we could call that instead of duplicating the switch statement.

<sup>1</sup>We can't add it to `package types` because that would create a circular dependency.

Reply:
Can we do this in a subsequent pull request?

Reply:
Added jira task